### PR TITLE
Rework "find_images_with_included_rpms" to use Pyxis APIs

### PR DESCRIPTION
**Pull request by ronnll**

### CODE REVIEW

File: freshmaker/lightblue.py [hunk 1]
```diff
@@ -996,21 +996,23 @@ def find_images_with_included_rpms(
...
 
-        image_request = {
-            "objectType": "containerImage",
-            "query": {},   # set by _set_container_image_filters()
-            "projection": self._get_default_projection(
-                rpm_names=rpm_name_to_nvrs.keys(),
-                include_rpm_manifest=include_rpm_manifest)
-        }
+        rpm_names = [kobo.rpmlib.parse_nvr(nvr)["name"] for nvr in rpm_nvrs]
+        images = self.pyxis.find_images_by_installed_rpms(rpm_name_to_nvrs.keys(), rpm_names, content_sets, repositories, published)
```
Comment (qixiang) [inline freshmaker/lightblue.py:1000]: Check the definition:
```
  def find_images_by_installed_rpms(                                                                                                                                                               
      self, rpm_names, content_sets=None, repositories=None, published=None, tags=None
  ):
```

File: tests/test_lightblue.py [hunk 2]
```diff
@@ -1317,11 +1388,10 @@ def test_find_images_with_included_srpm(self, exists, cont_images):
             repo["repository"]: repo for repo in
             self.fake_repositories_with_content_sets}
         # Add a duplicate simulating a multi-arch image
-        self.fake_container_images.append(self.fake_container_images[1])
-        cont_images.return_value = self.fake_container_images
+        #self.fake_container_images.append(self.fake_container_images[1])
+        #cont_images.return_value = self.fake_container_images
```
Comment (qixiang) [inline tests/test_lightblue.py:1392]: Please remove such unneeded code.

File: freshmaker/lightblue.py [hunk 3]
```diff
@@ -1873,3 +1874,4 @@ def get_fixed_published_image(self, name, version, image_group, rpm_nvrs, conten
         image = images[0]
         image.resolve(self)
         return image
+
```
Comment (qixiang) [inline freshmaker/lightblue.py:1877]: Please avoid such unnecessary lines, flake8 can report the warnings.

File: tests/test_lightblue.py [hunk 4]
```diff
@@ -1309,6 +1313,73 @@ def test_find_all_repositories(self, exists, gql_client):
...
+                            "tags": [{"name": "latest"}],
+                            }
+                        ],
+                    },
+        ],
+        "error": None,
+        "page": 0,
+        "page_size": 250,
+        "total": 2,
+    }}
```
Comment (qixiang) [inline tests/test_lightblue.py:1382]: Please fix the indention.

File: tests/test_lightblue.py [hunk 5]
```diff
@@ -1359,32 +1429,160 @@ def test_find_images_with_included_srpm(self, exists, cont_images):
         # in repository "product/repo1".
         self.assertEqual(ret, [cont_images.return_value[1]])
 
-    @patch('freshmaker.lightblue.LightBlue.find_container_images')
+    @mock.patch('freshmaker.pyxis_gql.Client')
```
Comment (qixiang) [inline tests/test_lightblue.py:1432]:
```suggestion
    @patch('freshmaker.pyxis_gql.Client')
```

File: tests/test_lightblue.py [hunk 6]
```diff
@@ -1359,32 +1429,160 @@ def test_find_images_with_included_srpm(self, exists, cont_images):
...
         repositories = {
             repo["repository"]: repo for repo in
             self.fake_repositories_with_content_sets}
-        cont_images.return_value = (
-            self.fake_container_images +
-            self.fake_container_images_floating_tag)
         ret = lb.find_images_with_included_rpms(
-            ["dummy-content-set-1", "dummy-content-set-2"], ["openssl-1.2.3-2"], repositories)
-
+            ["dummy-content-set-1", "dummy-content-set-2"], ["openssl-1.2.4-2"], repositories)
```
Comment (qixiang) [inline tests/test_lightblue.py:1511]: Check the method definition:
```
def find_images_by_installed_rpms(self, rpm_names, content_sets=None, repositories=None, published=None, tags=None):
```

File: tests/test_lightblue.py [hunk 7]
```diff
@@ -1359,32 +1429,160 @@ def test_find_images_with_included_srpm(self, exists, cont_images):
...
         self.assertEqual(
             [image.nvr for image in ret],
-            ['package-name-2-4-12.10', 'package-name-3-4-12.10'])
+            ['parent-1-3', 'parent-1-2'])
 
-    @patch('freshmaker.lightblue.LightBlue.find_container_images')
+    @mock.patch('freshmaker.pyxis_gql.Client')
```
Comment (qixiang) [inline tests/test_lightblue.py:1516]:
```suggestion
    @patch('freshmaker.pyxis_gql.Client')
```

File: tests/test_lightblue.py [hunk 8]
```diff
@@ -1393,34 +1591,92 @@ def test_images_with_included_newer_srpm(
         repositories = {
             repo["repository"]: repo for repo in
             self.fake_repositories_with_content_sets}
-        cont_images.return_value = (
-            self.fake_container_images +
-            self.fake_container_images_floating_tag)
         ret = lb.find_images_with_included_rpms(
             ["content-set-1", "content-set-2"], ["openssl-1.2.3-1"], repositories)
```
Comment (qixiang) [inline tests/test_lightblue.py:1595]: As above, check the method definition to see what's wrong here.

File: tests/test_lightblue.py [hunk 9]
```diff
@@ -1393,34 +1591,92 @@ def test_images_with_included_newer_srpm(
...
         self.assertEqual(ret, [])
 
-    @patch('freshmaker.lightblue.LightBlue.find_container_images')
+    @mock.patch('freshmaker.pyxis_gql.Client')
```
Comment (qixiang) [inline tests/test_lightblue.py:1598]:
```suggestion
    @patch('freshmaker.pyxis_gql.Client')
```

**Review by qixiang — CHANGES_REQUESTED** [review]
If you run the flake8 test, there are a lot errors and warnings.

File: freshmaker/lightblue.py (hunk 1 above)
Comment (ronnll) [reply freshmaker/lightblue.py]: https://github.com/redhat-exd-rebuilds/freshmaker/pull/203/files#diff-9ccbae0ffc4e5b4c63410f2592bc4a2088f97c3b45ec55ea5a6f793258038248R1000
auto_rebuild_tags added

File: tests/test_lightblue.py (hunk 6 above)
Comment (ronnll) [reply tests/test_lightblue.py:1510]: Resolving this as discussed!

File: tests/test_lightblue.py (hunk 8 above)
Comment (ronnll) [reply tests/test_lightblue.py]: Resolving this as discussed!

**Comment by ElenaKarolinaSemanova** [comment]
lgtm

**Pull request closed without merging**